web-console/src/views/load-data-view/load-data-view.tsx

```diff
@@ -2967,7 +2968,7 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
 
     try {
       const resp = await axios.get(`/druid/indexer/v1/task/${initTaskId}`);
-      this.updateSpec(resp.data.payload);
+      this.updateSpec(cleanSpec(resp.data.payload));
```

Comment:
What do you think about adding a test for the data loader to verify a reopened spec can be resubmitted?

Reply:
like this: https://github.com/apache/druid/pull/9536/files#diff-ff16ebd3aaa7f975353b795b18c80c2dR104

Reply:
Yeah, that's a good unit test for the utility method.

I was thinking it'd also be nice to have something that ensures the load data view itself is behaving as expected (e.g., it should be calling `cleanSpec` in the right places).

Reply:
I ideally we would have a Druid endpoint that can output the same spec that was given to it: https://github.com/apache/druid/issues/7631
The console calling `cleanSpec` is really a workaround around not having ☝️